cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectBuildUtils.java

```diff
@@ -21,9 +22,9 @@ public static boolean buildPodComplete(Pod pod)   {
                 && pod.getStatus() != null
                 && pod.getStatus().getContainerStatuses() != null
                 && pod.getStatus().getContainerStatuses().size() > 0
-                && pod.getStatus().getContainerStatuses().get(0) != null
-                && pod.getStatus().getContainerStatuses().get(0).getState() != null
-                && pod.getStatus().getContainerStatuses().get(0).getState().getTerminated() != null;
+                && getConnectBuildContainerStatus(pod) != null
+                && getConnectBuildContainerStatus(pod).getState() != null
+                && getConnectBuildContainerStatus(pod).getState().getTerminated() != null;
```

Comment:
I think you should try to avoid streaming through the conditions 3 times. Maybe make the `getConnectBuildContainerStatus` method smarter to to check the null and return the state or you can split the if and get the container status once and then use it multiple times?

Reply:
Yeah, makes totally sense. This should be fixed now. Thanks :)